### PR TITLE
Fix IK example in the documentation

### PR DESCRIPTION
**Pull request by stephane-caron**
This correction makes sure the error is in the local frame (previously in target frame).

Note that it reverts https://github.com/stack-of-tasks/pinocchio/pull/1875 which was not the correct fix to the convergence issue. Now we are back to the original target and the IK example converges correctly.

See https://github.com/stack-of-tasks/pinocchio/issues/1959#issuecomment-1543725756

### CODE REVIEW

**Comment by hrp2-14** [comment]
Hi ! This project doesn't usually accept pull requests on the main branch.
If this wasn't intentionnal, you can change the base branch of this PR to devel
(No need to close it for that). Best, a bot.

**Comment by stephane-caron** [comment]
Check log for the C++ example:

```console
pinocchio/build/examples$ ./example-cpp-inverse-kinematics
0: error = -1  0  1  0  0  0
10: error =   -0.431744           0   -0.296119           0 3.87165e-06           0
20: error =   -0.150299           0   -0.105353           0 1.27714e-06           0
30: error =  -0.0524834           0  -0.0369917           0 4.09649e-07           0
40: error =  -0.0183143           0  -0.0129315           0 1.29159e-07           0
50: error = -0.00638779           0 -0.00451311           0 4.01063e-08           0
60: error = -0.00222754           0 -0.00157414           0 1.22457e-08           0
70: error =  -0.00077673            0 -0.000548934            0  3.66118e-09            0
80: error = -0.000270834            0  -0.00019141            0  1.06406e-09            0
90: error = -9.44346e-05            0 -6.67416e-05            0  2.96875e-10            0
Convergence achieved!

result:        0 -6.28326        0  14.1373        0 -7.85404

final error: -7.64921e-05            0 -5.40608e-05            0  2.28456e-10            0
```

Check log for the Python example:

```console
$ python inverse-kinematics.py                                                                                                       ✔  pink  
0: error = [ 1.  0. -1.  0.  0.  0.]
10: error = [ 0.37879951  0.         -0.06820648  0.         -0.01549777  0.        ]
20: error = [ 0.13200257  0.         -0.02194841  0.         -0.00540374  0.        ]
30: error = [ 0.04602295  0.         -0.00742541  0.         -0.00188417  0.        ]
40: error = [ 0.01604754  0.         -0.00256131  0.         -0.00065697  0.        ]
50: error = [ 0.00559551  0.         -0.00088969  0.         -0.00022907  0.        ]
60: error = [ 1.95104597e-03  0.00000000e+00 -3.09806153e-04  0.00000000e+00
 -7.98720025e-05  0.00000000e+00]
70: error = [ 6.80289144e-04  0.00000000e+00 -1.07972737e-04  0.00000000e+00
 -2.78496453e-05  0.00000000e+00]
80: error = [ 2.37202341e-04  0.00000000e+00 -3.76416878e-05  0.00000000e+00
 -9.71057087e-06  0.00000000e+00]
Convergence achieved!

result: [0.0, -6.283277204116201, 0.0, 14.137244251306424, 0.0, -7.853963285116105]

final error: [ 9.18970703e-05  0.00000000e+00 -1.45823889e-05  0.00000000e+00
 -3.76207412e-06  0.00000000e+00]
```

File: examples/inverse-kinematics.cpp [hunk 1]
```diff
@@ -41,7 +41,10 @@ int main(int /* argc */, char ** /* argv */)
       success = false;
       break;
     }
-    pinocchio::computeJointJacobian(model,data,q,JOINT_ID,J);
+    pinocchio::computeJointJacobian(model,data,q,JOINT_ID,J);  // J in joint frame
+    pinocchio::Data::Matrix6x Jlog(6, 6);
```
Comment (jcarpent) [inline examples/inverse-kinematics.cpp:45]:
```suggestion
    pinocchio::Data::Matrix6 Jlog;
```

File: examples/inverse-kinematics.cpp [hunk 2]
```diff
@@ -41,7 +41,10 @@ int main(int /* argc */, char ** /* argv */)
...
+    pinocchio::Jlog6(iMd.inverse(), Jlog);
+    J = -Jlog * J
```
Comment (jcarpent) [inline examples/inverse-kinematics.cpp:47]:
```suggestion
    J = -Jlog * J;
```

**Comment by stephane-caron** [comment]
Hmm I don't see why the ROS melodic CI fails with:

```
107/117 Test #107: example-py-inverse-kinematics ......................***Failed    0.01 sec
```

This only happens in that CI.

@nim65s Maybe we can add ``--output-on-failure`` to CTest when running ``make test`` (as it currently only runs ``/usr/bin/ctest --force-new-ctest-process``)? I would have proposed a PR for this directly but looking at the CMakeLists I don't see where the ``ctest`` command is generated from (clearly not the ``run_tests`` custom target...).

**Comment by nim65s** [comment]
this can also be activated with the `CTEST_OUTPUT_ON_FAILURE=ON` environment variable. @wxmerkt : could you add it ?

File: examples/inverse-kinematics.py [hunk 3]
```diff
@@ -20,15 +20,16 @@
...
+    err = pinocchio.log(iMd).vector  # in joint frame
     if norm(err) < eps:
         success = True
         break
     if i >= IT_MAX:
         success = False
         break
-    J = pinocchio.computeJointJacobian(model,data,q,JOINT_ID)
+    J = pinocchio.computeJointJacobian(model,data,q,JOINT_ID)  # in joint frame
+    J = -pinocchio.Jlog6(iMd.inverse()) @ J
```
Comment (wxmerkt) [inline examples/inverse-kinematics.py:32]: @stephane-caron This is why ROS Melodic CI is failing - the `@` operator is Python3, Melodic is Python2.7. `np.dot` should fix it
```suggestion
    J = -np.dot(pinocchio.Jlog6(iMd.inverse()), J)
```

**Review by jcarpent — APPROVED** [review]

File: examples/inverse-kinematics.py (hunk 3 above)
Comment (nim65s) [reply examples/inverse-kinematics.py]: workarounded in #1966

**Pull request merged**